tests/e2e/pact/resources/common.ts

```diff
@@ -6,30 +6,103 @@ import {ExampleFile} from "../models/files/example-file"
+// models/examples with path separator replaced by space
+// or set pactGroups = [""] to run all together
+export const pactGroups = [
+  "secondary-care community acute",
+  "secondary-care community repeat-dispensing",
+  "secondary-care homecare",
+  "primary-care"
+] as const
+
+export const cancelPactGroups = [
+  "secondary-care community acute"
+] as const
 
-export const PactGroups = [
-  "accept-header",
-  "failures",
-  "secondarycare-community-acute",
-  "secondarycare-community-repeatdispensing",
-  "secondarycare-homecare",
-  "primarycare",
-  "secondarycare-community-acute-cancel",
+export const releasePactGroups = [
+  ""
 ]
-export type PactGroup = typeof PactGroups
 
-export function pactOptions(mode: ApiMode, endpoint: ApiEndpoint, group?: PactGroup): JestPactOptions {
+export const failurePactGroups = [
+  "failures"
+] as const
+
+export const miscPactGroups = [
+  "accept_headers"
+] as const
+
+export const allPactGroups = [...pactGroups, ...cancelPactGroups, ...failurePactGroups, ...miscPactGroups]
+export type AllPactGroups = typeof pactGroups[number] | typeof cancelPactGroups[number] | typeof failurePactGroups[number] | typeof miscPactGroups[number]
+
+export class PactGroupCases {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+  constructor(name: AllPactGroups, cases: any) {
+    this.name = name
+    this.cases = cases
+  }
+  name: AllPactGroups
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  cases: any
+}
+
+// used to add type-safety for adding a new pact
+export function pactOptions(mode: ApiMode, endpoint: ApiEndpoint, group?: AllPactGroups, operation?: ApiOperation): JestPactOptions
+{
   const sandbox = mode === "sandbox"
-return {
-      spec: 3,
+  const groupName = group?.replace(/-/g, "").replace(/\s/g, "-")
+  const operationName = operation === "send" ? "" : operation
+  return {
+    spec: 3,
     consumer: `nhsd-apim-eps-test-client+${process.env.PACT_VERSION}`,
-    provider: `nhsd-apim-eps${sandbox ? "-sandbox" : ""}+${endpoint}${group ? "-" + group : ""}+${process.env.PACT_VERSION}`,
-      pactfileWriteMode: "overwrite"
+    provider: `nhsd-apim-eps${sandbox ? "-sandbox" : ""}+${endpoint}${groupName ? "-" + groupName : ""}${operationName ? "-" + operationName : ""}+${process.env.PACT_VERSION}`,
+    pactfileWriteMode: "merge"
   }
 }
 
+// get pact groups for verification
+export const pactGroupNames = convertPactDescriptionsToPactNames(pactGroups)
+const releasePactGroupNames = convertPactDescriptionsToPactNames(releasePactGroups)
+const cancelPactGroupNames = convertPactDescriptionsToPactNames(cancelPactGroups)
+
+// convert pact group name from description search string format to single string
+// matching the published pact's name
+function convertPactDescriptionsToPactNames(descriptions: readonly string[]) {
+ return descriptions.map(g => g.replace(/-/g, "").replace(/\s/g, "-"))
```

Comment:
Might be good to create a method which does this replacement and use it in both places (here and line 56)

Reply:
Good shout